gammapy/datasets/map.py

```diff
@@ -2569,19 +2573,36 @@ def compute_flux_psf_convolved(self):
         """Compute psf convolved and temporal model corrected flux."""
         value = self.compute_flux_spectral()
 
-        if self.model.spatial_model and not isinstance(self.geom, RegionGeom):
-            value = value * self.compute_flux_spatial().quantity
+        if self.model.spatial_model:
+            value = value * self.compute_flux_spatial()#.quantity
 
         if self.model.temporal_model:
             value *= self.compute_temporal_norm()
 
         return Map.from_geom(geom=self.geom, data=value.value, unit=value.unit)
 
     def _compute_flux_spatial(self):
-        """Compute spatial flux"""
-        value = self.model.spatial_model.integrate_geom(self.geom)
-        if self.psf and self.model.apply_irf["psf"]:
-            value = self.apply_psf(value)
+        """Compute spatial flux
+
+        Returns
+        ----------
+        value: `~astropy.units.Quantity`
+            Psf-corrected, integrated flux over a given region.
+        """
+        if isinstance(self.geom, RegionGeom):
+            wcs_geom = self.geom.to_wcs_geom()
+            mask = self.geom.contains(wcs_geom.get_coord())
+            values = self.model.spatial_model.integrate_geom(wcs_geom)
+
+            if self.psf and self.model.apply_irf["psf"]:
+                values = self.apply_psf(values)
+            value = (values.data[mask]).sum()
```

Comment:
I started to test implementation and I noticed a bug in this line: we have to sum only over the spatial axes here and not all axes (and not including the energy axis). So the statement here should be something like `.sum(axis=(1, 2))`... I need to check whether the implementation in `SpatialModel.integrate_geom()` is correct.